### ISSUE TITLE
Windows Build error

### ISSUE THREAD

**Issue by jaseelmohd** [Mobile-Artificial-Intelligence/maid#320]
Thanks for the awesome repository.

I am running into an erro when I build for windows, I tried different release checkpoints in the repository, still I am not able to get it to run. Here is the error message that it says when I build for windows:

```
[        ]   C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(166,5): error MSB3073: The     
command "setlocal [D:\maid_working_version\build\windows\x64\INSTALL.vcxproj]
[        ] C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(166,5): error MSB3073:
"C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" -DBUILD_TYPE=Debug -P
cmake_install.cmake [D:\maid_working_version\build\windows\x64\INSTALL.vcxproj]
[        ] C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(166,5): error MSB3073: if        
%errorlevel% neq 0 goto :cmEnd [D:\maid_working_version\build\windows\x64\INSTALL.vcxproj]
[        ] C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(166,5): error MSB3073: :cmEnd
[D:\maid_working_version\build\windows\x64\INSTALL.vcxproj]
[        ] C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(166,5): error MSB3073: endlocal &
call :cmErrorLevel %errorlevel% & goto :cmDone [D:\maid_working_version\build\windows\x64\INSTALL.vcxproj]
[        ] C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(166,5): error MSB3073:
:cmErrorLevel [D:\maid_working_version\build\windows\x64\INSTALL.vcxproj]
[        ] C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(166,5): error MSB3073: exit /b %1
[D:\maid_working_version\build\windows\x64\INSTALL.vcxproj]
[        ] C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(166,5): error MSB3073: :cmDone   
[D:\maid_working_version\build\windows\x64\INSTALL.vcxproj]
[        ] C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(166,5): error MSB3073: if        
%errorlevel% neq 0 goto :VCEnd [D:\maid_working_version\build\windows\x64\INSTALL.vcxproj]
[        ] C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(166,5): error MSB3073: :VCEnd"   
exited with code 1. [D:\maid_working_version\build\windows\x64\INSTALL.vcxproj]
[        ]     15 Warning(s)
[        ]     1 Error(s)
[        ] Time Elapsed 00:01:00.23
[  +35 ms] Building Windows application... (completed in 77.2s)
[   +1 ms] Exception: Build process failed.
[   +5 ms] "flutter run" took 80,913ms.
[   +3 ms]
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
           #1      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:790:9)
           <asynchronous suspension>
           #2      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1367:27)
           <asynchronous suspension>
           #3      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #4      CommandRunner.runCommand (package:args/command_runner.dart:212:13)
           <asynchronous suspension>
           #5      FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:348:9)
           <asynchronous suspension>
           #6      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #7      FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:294:5)
           <asynchronous suspension>
           #8      run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:112:9)
           <asynchronous suspension>
           #9      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #10     main (package:flutter_tools/executable.dart:90:3)
           <asynchronous suspension>


[ +158 ms] ensureAnalyticsSent: 155ms
[        ] Running 1 shutdown hook
[  +10 ms] Shutdown hooks complete
[        ] exiting with code 1
```

Please let me know if I have done something wrong, I am unable to figure out where I am going wrong. Appreciate the help

**Comment by danemadsen** [comment]
yeah windows seems to be the most temperamental platform so far. 

Just to confirm though are you cloning the sub-modules? Maid wont compile if the sub-modules aren't cloned. You can clone maid with its sub-modules by running ```git clone --recursive https://github.com/Mobile-Artificial-Intelligence/maid.git```, or if you've already cloned maid you can run ```get submodule init``` and ```git submodule update``` to fetch the submodules.

I may stop using the submodules in the future but for the time being they're required

**Comment by jaseelmohd** [comment]
> yeah windows seems to be the most temperamental platform so far.
> 
> Just to confirm though are you cloning the sub-modules? Maid wont compile if the sub-modules aren't cloned. You can clone maid with its sub-modules by running `git clone --recursive https://github.com/Mobile-Artificial-Intelligence/maid.git`, or if you've already cloned maid you can run `get submodule init` and `git submodule update` to fetch the submodules.
> 
> I may stop using the submodules in the future but for the time being they're required

Yes, I am cloning it using the recursive tag. Tried it again to just confirm, and still shows the error

**Comment by danemadsen** [comment]
> > yeah windows seems to be the most temperamental platform so far.
> > Just to confirm though are you cloning the sub-modules? Maid wont compile if the sub-modules aren't cloned. You can clone maid with its sub-modules by running `git clone --recursive https://github.com/Mobile-Artificial-Intelligence/maid.git`, or if you've already cloned maid you can run `get submodule init` and `git submodule update` to fetch the submodules.
> > I may stop using the submodules in the future but for the time being they're required
> 
> Yes, I am cloning it using the recursive tag. Tried it again to just confirm, and still shows the error

Hmm, ok ill have to look into it? what version of windows are you attempting to build on? 10? 11?

**Comment by danemadsen** [comment]
and your cmake version? (run ```cmake --version```)

Also is there anything that shows up missing when you run ```flutter doctor```

Sorry for the interrogation but i need this info to diagnose

**Comment by jaseelmohd** [comment]
cmake version 3.28.1

Flutter doctor does not show any issues:
```
[✓] Flutter (Channel stable, 3.16.9, on Microsoft Windows [Version 10.0.22631.3085], locale en-US)
    • Flutter version 3.16.9 on channel stable at C:\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 41456452f2 (12 days ago), 2024-01-25 10:06:23 -0800
    • Engine revision f40e976bed
    • Dart version 3.2.6
    • DevTools version 2.28.5

[✓] Windows Version (Installed version of Windows is version 10 or higher)

[!] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at C:\Users\jazee\AppData\Local\Android\Sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/windows#android-setup for more details.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.8.6)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.8.34525.116
    • Windows 10 SDK version 10.0.22621.0

[✓] Android Studio (version 2023.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-b2043.56-10550314)

[✓] IntelliJ IDEA Community Edition (version 2023.2)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2023.1.3
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.86.0)
    • VS Code at C:\Users\jazee\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.82.0

[✓] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22631.3085]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 121.0.6167.140
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 121.0.2277.98

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.`
```

Your fine, if you need any more info let me know.

**Comment by danemadsen** [comment]
I cant reproduce it. what commit are you on? latest?

**Comment by jaseelmohd** [comment]
Hey @danemadsen ,
Sorry for the late response. I am currently using the 1.1.6 version of the app (commit #236). I am using this commit because it is currently the most stable ones that work with my models. Though, I tested other commits and still get the same error.

Also, I found a workaround to make the windows build work, this is not ideal approach but here it is:
1. Delete windows folder, and recreate it using `flutter create --platform=windows .`
2. Then build the windows app, it will be successful most definetely
3. Copy thr llama.cpp files from the pre-built release zip available in the repository.

This is how I did, and it works with both debug and release builds, as far as I know the issue is with building both the llama files and the windows app together is causing some issues.
